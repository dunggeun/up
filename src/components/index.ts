export * from './FadeInView';
export * from './KeyboardAvoidingView';
export * from './SafeAreaView';
export * from './Modal';
export * from './PageContent';
export * from './PageTitle';
export * from './ListItem';
export * from './Section';
export * from './TabBar';
export * from './Toast';
export * from './TransitionGroup';
