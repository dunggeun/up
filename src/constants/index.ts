import { Dimensions, Platform } from 'react-native';
import { version, dependencies } from '../../package.json';

export const VERSION = version;
export const DEPENDENCIES = dependencies;

export const APP_MINIMUM_LOADING_DURATION = 1000;

export const CONTAINER_MAX_WIDTH = 600;
export const APP_BAR_HEIGHT = 56;

export const LANDING_LOGO_SIZE = 180;
export const LANDING_LOGO_MARGIN = 100;

export const HIT_SLOP = 10;
export const TOUCHABLE_OPACITY_HIT_SLOP = {
  top: HIT_SLOP,
  bottom: HIT_SLOP,
  left: HIT_SLOP,
  right: HIT_SLOP,
} as const;

export const BORDER_WIDTH = 2;
export const PRESSABLE_DEPTH = 5;

export const TOAST_ANIMATION_DURATION = 500;
export const TOAST_DURATION = 3000;

const window = Dimensions.get('window');
export const WINDOW_WIDTH = window.width;
export const WINDOW_HEIGHT = window.height;

export const DEVELOPER_EMAIL = 'dev.ghlee@gmail.com';
export const UP_REPOSITORY_URL = 'https://github.com/dunggeun/up';

export const SHARED_CONFIG = {
  navigatorCardStyle: {
    flex: 1,
    backgroundColor: 'white',
  },
  scrollableViewProps: {
    onStartShouldSetResponderCapture: () => false,
    onMoveShouldSetResponderCapture: () => false,
  },
} as const;

export const IS_NATIVE = Platform.OS === 'android' || Platform.OS === 'ios';
export const IS_WEB = Platform.OS === 'web';
export const USE_NATIVE_DRIVER = IS_NATIVE;

export const BACKUP_FILE_NAME = 'up.bak';

export const SENTRY_RELEASE = `up@${VERSION}${__DEV__ ? '-dev' : ''}`;
export const SENTRY_SAMPLE_RATE = __DEV__ ? 1.0 : 0.25;
