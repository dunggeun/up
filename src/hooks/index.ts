export * from './useDebounce';
export * from './useIsAuthorized';
export * from './useMainTabBarInset';
export * from './usePromise';
export * from './useUserThemeColor';
