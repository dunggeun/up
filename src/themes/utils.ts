import { Platform } from 'react-native';

export const webFont = (font: string): string => {
  return Platform.select({
    web: `${font}, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, Inter-serif`,
    default: font,
  });
};
