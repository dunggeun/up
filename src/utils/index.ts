export * from './async';
export * from './common';
export * from './locales';
