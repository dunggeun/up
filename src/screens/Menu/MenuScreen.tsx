import React, { useState } from 'react';
import { useActor } from '@xstate/react';
import { FadeInView } from 'src/components';
import { delay } from 'src/utils';
import { AppManager } from 'src/modules';
import { APP_MINIMUM_LOADING_DURATION } from 'src/constants';
import { DeleteConfirmModal } from './components';
import { Menu } from './Menu';

import type { MainTabProps } from 'src/navigators/MainTab/types';

type MenuProps = MainTabProps<'Menu'>;

const app = AppManager.getInstance();

export function MenuScreen(_props: MenuProps): JSX.Element {
  const [
    deleteConfirmModalVisibility,
    setDeleteConfirmModalVisibility
  ] = useState(false);
  const [loading, setLoading] = useState(false);
  const [_, send] = useActor(app.getService());

  const handlePressReset = (): void => {
    setDeleteConfirmModalVisibility(true);
  };

  const handleCloseDeleteConfirmModal = (): void => {
    setDeleteConfirmModalVisibility(false);
  };

  const handleDelete = async (): Promise<void> => {
    if (loading) return;

    setLoading(true);

    try {
      await Promise.all([
        app.reset(),
        delay(APP_MINIMUM_LOADING_DURATION),
      ]);
      setDeleteConfirmModalVisibility(false);
      send({ type: 'REFRESH' });
    } catch (error) {
      // @todo: error handing
    } finally {
      setLoading(false);
    }
  };

  return (
    <FadeInView>
      <Menu onPressReset={handlePressReset}/>
      <DeleteConfirmModal
        isLoading={loading}
        onClose={handleCloseDeleteConfirmModal}
        onDelete={(): void => void handleDelete()}
        visible={deleteConfirmModalVisibility}
      />
    </FadeInView>
  );
}
