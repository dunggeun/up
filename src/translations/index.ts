/* eslint-disable import/no-named-as-default-member */

import i18next, {
  type LanguageDetectorModule,
  type InitOptions,
} from 'i18next';
import { initReactI18next } from 'react-i18next';
import { noop, getCurrentLocale } from 'src/utils';
import en from './en.json';
import ko from './ko.json';

const languageDetector: LanguageDetectorModule = {
  type: 'languageDetector',
  detect: () => getCurrentLocale(),
  init: noop,
  cacheUserLanguage: noop,
} as const;

const options: InitOptions = {
  compatibilityJSON: 'v3',
  fallbackLng: 'en',
  resources: { ko, en },
  ns: ['common'],
  defaultNS: 'common',
  interpolation: {
    escapeValue: false,
  },
} as const;

i18next.use(languageDetector).use(initReactI18next).init(options).catch(noop);

export const i18n = i18next;
export const t = (key: string): string => i18next.t(key);
