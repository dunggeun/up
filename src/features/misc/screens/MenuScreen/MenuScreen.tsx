import React, { useState, useMemo, useCallback, useEffect } from 'react';
import { InteractionManager } from 'react-native';
import DateTimePickerModal from 'react-native-modal-datetime-picker';
import { useActor } from '@xstate/react';
import dayjs from 'dayjs';
import { Pressable, styled, useDripsyTheme, View } from 'dripsy';
import { useRecoilValue } from 'recoil';
import { useUserThemeColor, type User } from 'src/features/users';
import { AppEventChannel } from 'src/modules/event';
import { requestNotificationPermission } from 'src/modules/notifications';
import { navigate } from 'src/navigators/helpers';
import { globalMachineService } from 'src/stores/machines';
import { notificationStatus as notificationStatusAtom } from 'src/stores/recoil';
import { useMainTabBarInset } from 'src/hooks';
import { openMail, openUrl, runAfterModalDismissed } from 'src/utils';
import {
  DEFAULT_REMINDER_TIME,
  DEVELOPER_EMAIL,
  IS_NATIVE,
  REMINDER_TIME_FORMAT,
  VERSION,
} from 'src/constants';
import { CommonLayout, H2, Toggle } from 'src/designs';
import { FadeInView, ListItem } from 'src/components';
import { t } from 'src/translations';
import { replacePlaceholder } from '../../../../utils/string';
import type { MainTabProps } from 'src/navigators/MainTab/types';

type MenuProps = MainTabProps<'Menu'>;

const Option = styled(Pressable)({
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
  paddingY: '$04',
});

export function MenuScreen(_props: MenuProps): React.ReactElement {
  const [timePickerVisibility, setTimePickerVisibility] = useState(false);
  const userColor = useUserThemeColor();
  const { theme } = useDripsyTheme();
  const { bottomInset } = useMainTabBarInset();
  const [state, send] = useActor(globalMachineService);
  const notificationStatus = useRecoilValue(notificationStatusAtom);
  const remindAt = state.context.user?.remindAt;
  const remindAtAsDate = useMemo(
    () => dayjs(remindAt, REMINDER_TIME_FORMAT).toDate(),
    [remindAt],
  );
  const hapticEnabled = state.context.user?.settings.enableHaptic ?? false;
  const canUseReminder =
    notificationStatus.error === null && notificationStatus.granted;

  useEffect(() => {
    if (notificationStatus.granted) return;
    requestNotificationPermission();
  }, [notificationStatus.granted]);

  const handlePressGuide = (): void => {
    openUrl(t('url.guide'));
  };

  const handlePressDataManagement = (): void => {
    navigate('Common', 'DataManagement');
  };

  const handlePressSendFeedback = (): void => {
    openMail(DEVELOPER_EMAIL, {
      subject: t('template.send_feedback.title'),
      body: t('template.send_feedback.body'),
    });
  };

  const handlePressOpenSource = (): void => {
    navigate('Common', 'OpenSourceProject');
    InteractionManager.runAfterInteractions(() => {
      AppEventChannel.getInstance().dispatch('enterOpenSource', undefined);
    });
  };

  const handleChangeReminderToggle = useCallback(
    (value: boolean): void => {
      send({
        type: 'EDIT_USER',
        user: {
          remindAt: value ? DEFAULT_REMINDER_TIME : null,
        } as Partial<User>,
      });
    },
    [send],
  );

  const handleCancelRemindModal = (): void => {
    setTimePickerVisibility(false);
  };

  const handleConfirmRemindTime = (date: Date): void => {
    setTimePickerVisibility(false);
    runAfterModalDismissed(() => {
      send({
        type: 'EDIT_USER',
        user: {
          remindAt: dayjs(date).format(REMINDER_TIME_FORMAT),
        } as Partial<User>,
      });
    });
  };

  const handleChangeHapticFeedbackToggle = useCallback(
    (value: boolean): void => {
      send({
        type: 'EDIT_USER',
        user: {
          settings: { enableHaptic: value },
        } as Partial<User>,
      });
    },
    [send],
  );

  const renderNativeOptions = (): React.ReactElement | null => {
    if (!IS_NATIVE) return null;

    const reminderEnabledLabel = remindAt
      ? replacePlaceholder(t('label.reminder_time'), remindAt)
      : t('label.enable_reminder');

    return (
      <>
        <Option onPress={(): void => setTimePickerVisibility(true)}>
          <H2 variant="primary">
            {canUseReminder
              ? reminderEnabledLabel
              : t('message.error.permission_required')}
          </H2>
          <Toggle
            color={userColor}
            disabled={!canUseReminder}
            initialValue={Boolean(remindAt)}
            onChange={handleChangeReminderToggle}
          />
        </Option>
        <Option disabled>
          <H2 variant="primary">{t('label.haptic_feedback')}</H2>
          <Toggle
            color={userColor}
            initialValue={hapticEnabled}
            onChange={handleChangeHapticFeedbackToggle}
          />
        </Option>
      </>
    );
  };

  return (
    <>
      <FadeInView>
        <CommonLayout insetBottom={false}>
          <CommonLayout.Header title={t('title.menu')} />
          <CommonLayout.Body>
            <ListItem label={t('label.version')} subLabel={VERSION} />
            <ListItem label={t('label.guide')} onPress={handlePressGuide} />
            {renderNativeOptions()}
            <ListItem
              label={t('label.data_management')}
              onPress={handlePressDataManagement}
            />
            <ListItem
              label={t('label.send_feedback')}
              onPress={handlePressSendFeedback}
            />
            <ListItem
              label={t('label.open_source')}
              onPress={handlePressOpenSource}
            />
            <View sx={{ height: bottomInset }} />
          </CommonLayout.Body>
        </CommonLayout>
      </FadeInView>
      <DateTimePickerModal
        accentColor={theme.colors[userColor]}
        cancelTextIOS={t('label.cancel')}
        confirmTextIOS={t('label.ok')}
        date={remindAtAsDate}
        is24Hour
        isDarkModeEnabled={false}
        isVisible={timePickerVisibility}
        locale="en_EN"
        mode="time"
        onCancel={handleCancelRemindModal}
        onConfirm={handleConfirmRemindTime}
        textColor={theme.colors.$text_primary}
      />
    </>
  );
}
