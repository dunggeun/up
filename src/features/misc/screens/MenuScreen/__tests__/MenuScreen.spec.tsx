import React from 'react';
import {
  render,
  screen,
  cleanup,
  fireEvent,
} from '@testing-library/react-native';
import { withDripsy } from 'tests/utils';
import { t } from 'src/translations';

import { MenuScreen } from '../MenuScreen';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const getMockedProps = (): any => ({
  navigation: {
    goBack: jest.fn(),
  },
});

describe('screens/MenuScreen', () => {
  afterEach(() => {
    cleanup();
  });

  describe('렌더링 되었을 때', () => {
    it('스냅샷이 일치해야 한다', () => {
      const tree = render(
        withDripsy(<MenuScreen {...getMockedProps()} />),
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('데이터 초기화 버튼을 눌렀을 때', () => {
    beforeEach(() => {
      const props = getMockedProps();
      render(withDripsy(<MenuScreen {...props} />));
      const resetButton = screen.getByText(t('label.reset_data'));
      fireEvent(resetButton, 'press');
    });

    it('삭제 확인 모달이 노출되어야 한다', () => {
      const deleteConfirmModalContent = screen.getByTestId(
        'delete-confirm-modal',
      );
      expect(deleteConfirmModalContent).not.toBeNull();
    });
  });
});
