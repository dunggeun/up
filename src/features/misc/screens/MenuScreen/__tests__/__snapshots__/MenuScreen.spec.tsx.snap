// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/MenuScreen 렌더링 되었을 때 스냅샷이 일치해야 한다 1`] = `
<View
  collapsable={false}
  style={
    {
      "flex": 1,
      "opacity": 0,
    }
  }
>
  <View
    style={
      [
        {},
        {
          "flex": 1,
        },
        undefined,
        {
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
        {},
        {},
      ]
    }
  >
    <View
      style={
        [
          {
            "alignSelf": "center",
            "backgroundColor": "#ffffff",
            "flex": 1,
            "marginHorizontal": "auto",
            "maxWidth": 600,
            "paddingHorizontal": 16,
            "width": "100%",
          },
          undefined,
          {},
        ]
      }
    >
      <View
        style={
          [
            {},
            {
              "marginHorizontal": -16,
              "zIndex": 3,
            },
            undefined,
            {},
            {},
          ]
        }
      >
        <View
          style={
            [
              {},
              {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "flexDirection": "row",
                "height": 56,
                "justifyContent": "space-between",
                "paddingHorizontal": 16,
                "position": "relative",
              },
              undefined,
              {},
              {},
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "bottom": 0,
                  "elevation": 3,
                  "height": 16,
                  "left": 0,
                  "marginBottom": -16,
                  "position": "absolute",
                  "right": 0,
                  "zIndex": 3,
                },
                undefined,
                {},
                {},
              ]
            }
          >
            <RNSVGSvgView
              align="none"
              bbHeight="100%"
              bbWidth="100%"
              focusable={false}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              vbHeight={100}
              vbWidth={100}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGDefs>
                  <RNSVGLinearGradient
                    gradient={
                      [
                        0,
                        -1,
                        1,
                        16777215,
                      ]
                    }
                    gradientTransform={
                      [
                        6.123233995736766e-17,
                        1,
                        -1,
                        6.123233995736766e-17,
                        0,
                        0,
                      ]
                    }
                    gradientUnits={1}
                    name="g"
                    x1="0"
                    x2="100"
                    y1="0"
                    y2="0"
                  />
                </RNSVGDefs>
                <RNSVGRect
                  fill={
                    {
                      "brushRef": "g",
                      "type": 1,
                    }
                  }
                  height="100%"
                  propList={
                    [
                      "fill",
                    ]
                  }
                  width="100%"
                  x="0"
                  y="0"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              [
                {},
                {
                  "height": 30,
                  "width": 30,
                },
                undefined,
                {},
                {},
              ]
            }
          />
          <Text
            accessibilityHint="제목"
            accessibilityLabel="메뉴"
            accessibilityRole="header"
            style={
              [
                {
                  "fontSize": 48,
                  "fontWeight": "bold",
                  "marginVertical": 26.56,
                },
                [
                  {
                    "fontFamily": "BMJUA",
                    "fontSize": 24,
                    "marginVertical": 0,
                    "paddingBottom": 4,
                  },
                  {
                    "fontWeight": "normal",
                  },
                  undefined,
                  {
                    "color": "#2e2e2e",
                  },
                ],
              ]
            }
          >
            메뉴
          </Text>
          <View
            style={
              [
                {},
                {
                  "height": 30,
                  "width": 30,
                },
                undefined,
                {},
                {},
              ]
            }
          />
        </View>
      </View>
      <RCTScrollView
        collapsable={false}
        onMoveShouldSetResponderCapture={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        scrollEventThrottle={0.0001}
        style={
          {
            "flex": 1,
            "marginHorizontal": -16,
            "paddingHorizontal": 16,
          }
        }
      >
        <View>
          <View
            style={
              [
                {},
                {
                  "flex": 1,
                  "gap": 16,
                },
                undefined,
                {},
                {},
              ]
            }
          >
            <View
              style={
                [
                  {},
                  {},
                  undefined,
                  {},
                  {},
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  style={
                    [
                      {
                        "fontSize": 48,
                        "fontWeight": "bold",
                        "marginVertical": 26.56,
                      },
                      [
                        {
                          "color": "#2e2e2e",
                          "fontFamily": "BMJUA",
                          "fontSize": 24,
                          "marginVertical": 0,
                          "paddingBottom": 4,
                        },
                        {
                          "fontWeight": "normal",
                        },
                        undefined,
                        {},
                      ],
                    ]
                  }
                >
                  버전
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "#777777",
                        "fontFamily": "BMJUA",
                        "fontSize": 16,
                        "paddingBottom": 4,
                      },
                      undefined,
                      {},
                    ]
                  }
                >
                  0.0.1
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  style={
                    [
                      {
                        "fontSize": 48,
                        "fontWeight": "bold",
                        "marginVertical": 26.56,
                      },
                      [
                        {
                          "color": "#2e2e2e",
                          "fontFamily": "BMJUA",
                          "fontSize": 24,
                          "marginVertical": 0,
                          "paddingBottom": 4,
                        },
                        {
                          "fontWeight": "normal",
                        },
                        undefined,
                        {},
                      ],
                    ]
                  }
                >
                  피드백 보내기
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  style={
                    [
                      {
                        "fontSize": 48,
                        "fontWeight": "bold",
                        "marginVertical": 26.56,
                      },
                      [
                        {
                          "color": "#2e2e2e",
                          "fontFamily": "BMJUA",
                          "fontSize": 24,
                          "marginVertical": 0,
                          "paddingBottom": 4,
                        },
                        {
                          "fontWeight": "normal",
                        },
                        undefined,
                        {},
                      ],
                    ]
                  }
                >
                  UP 평점주기 & 리뷰
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  style={
                    [
                      {
                        "fontSize": 48,
                        "fontWeight": "bold",
                        "marginVertical": 26.56,
                      },
                      [
                        {
                          "color": "#2e2e2e",
                          "fontFamily": "BMJUA",
                          "fontSize": 24,
                          "marginVertical": 0,
                          "paddingBottom": 4,
                        },
                        {
                          "fontWeight": "normal",
                        },
                        undefined,
                        {},
                      ],
                    ]
                  }
                >
                  오픈소스 프로젝트
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingVertical": 16,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontFamily": "BMJUA",
                        "fontSize": 16,
                        "paddingBottom": 4,
                      },
                      undefined,
                      {
                        "color": "#cccccc",
                      },
                    ]
                  }
                >
                  데이터 초기화
                </Text>
              </View>
              <View
                style={
                  [
                    {},
                    undefined,
                    {
                      "height": 102,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
</View>
`;
